### ISSUE TITLE
[combobox] accessibility: Guidance text not announced by screen readers

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4250]
# Bug report

## Current behavior

The "Async search" variant reveals instruction text on ArrowDown, which is not announced by NVDA, JAWS or VoiceOver. 

![The combobox input with text that has appeared below saying "Start typing to search people..."](https://github.com/user-attachments/assets/41a7c3b4-57c0-40a7-a676-48f10f4e521d)

The `aria-live` region looks like it is being dynamically rendered instead of being dynamically populated.

## Expected behavior

When instructional text appears for an input, this must be announced by screen readers.

## Reproducible example

Steps to replicate:
1. Turn on a screen reader
2. Navigate to [Combobox: Async search example](https://base-ui.com/react/components/combobox#async-search-single)
3. When focused on the input, use ArrowDown to make the instruction text appear visually
4. Note it is not announced by the screen reader

## Base UI version

v1.2.0

## Which browser are you using?

Chrome

## Which OS are you using?

Windows and MacOS

## Which assistive tech are you using (if applicable)?

NVDA, JAWS, VoiceOver

## Additional context

Using `role="alert"` on the guidance text's container or pre-rendering the aria-live region should fix the announcements.

Adding an aria-describedby to the input may also improve announcements.
